src/main/java/de/srendi/advancedperipherals/common/items/WeakMechanicSoul.java

```diff
@@ -0,0 +1,104 @@
+package de.srendi.advancedperipherals.common.items;
+
+import de.srendi.advancedperipherals.common.setup.Items;
+import de.srendi.advancedperipherals.common.util.EnumColor;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.monster.EndermanEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Hand;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.FakePlayer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class WeakMechanicSoul extends APItem {
+
+    public static class MechanicalSoulRecord {
+        public final int requiredCount;
+        public final Class<? extends Entity> entityClass;
+        public final Item resultSoul;
+
+        public MechanicalSoulRecord(int requiredCount, Class<? extends Entity> entityClass, Item resultSoul) {
+            this.requiredCount = requiredCount;
+            this.entityClass = entityClass;
+            this.resultSoul = resultSoul;
+        }
+    }
+
+    // So, I have
```

Comment:
Mh?

Reply:
Oh, will fix now) This is not about code below